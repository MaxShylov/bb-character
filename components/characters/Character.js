import { useCallback } from 'react';
import { TouchableOpacity, Text, View, StyleSheet } from 'react-native';
import { Avatar } from '../common';

export const Character = ({ data, onPressItem: handlePressItem }) => {
  const handlePress = useCallback(
    () => handlePressItem && handlePressItem(data),
    [data, handlePressItem]
  )

  return (
    <TouchableOpacity onPress={handlePress}>
      <View style={styles.wrapper}>
        <Avatar uri={data?.image} />
        <Text style={styles.text}>{data?.name}</Text>
      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 20,
  },
  text: {
    marginLeft: 20,
    fontSize: 24
  }
})
