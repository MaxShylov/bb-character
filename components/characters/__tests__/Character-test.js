import renderer from 'react-test-renderer';
import { Character } from '../Character';

test('renders correctly', () => {
  const tree = renderer.create(<Character />).toJSON();

  expect(tree).toMatchSnapshot();
});
