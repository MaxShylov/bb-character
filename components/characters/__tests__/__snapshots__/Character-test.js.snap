// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 20,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "borderColor": "#ccc",
            "borderRadius": 35,
            "borderWidth": 2,
            "height": 70,
            "width": 70,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
          "marginLeft": 20,
        }
      }
    />
  </View>
</View>
`;
